### PR TITLE
Support time events and payload attributes

### PR DESCRIPTION
**Pull request by krzysztofreczek**

### CODE REVIEW

File: tracing.go [hunk 1]
```diff
@@ -70,6 +77,16 @@ func TracingMiddleware(h message.HandlerFunc) message.HandlerFunc {
 			span.End()
 		}()
 
+		eIDString := msg.Metadata.Get(spanEventIDKey)
+		eID, _ := strconv.ParseInt(eIDString, 10, 64)
+
+		messageBytes := []byte(msg.Payload)
```
Comment (roblaszczak) [inline tracing.go]: As far as I remember it's alrady []byte :P

File: tracing.go [hunk 2]
```diff
@@ -2,15 +2,22 @@
...
 	"github.com/ThreeDotsLabs/watermill/message"
 	"go.opencensus.io/trace"
 	"go.opencensus.io/trace/propagation"
 )
 
-const spanContextKey = "opencensus_span_context"
+const (
+	spanContextKey               = "opencensus_span_context"
+	spanEventIDKey               = "opencensus_event_id"
+	spanEventPayloadAttributeKey = "event-payload"
```
Comment (czeslavo) [inline tracing.go]: Is there a reason for using mixed convention for key names (xxx-xxx vs xxx_xxx)?

File: tracing.go [hunk 3]
```diff
@@ -111,13 +128,38 @@ type publisherDecorator struct {
...
+		span.AddMessageSendEvent(eID, int64(messageBytesSize), 0)
+
+		SetSpanContext(span.SpanContext(), msg)
 	}
 
 	return d.Publisher.Publish(topic, messages...)
 }
+
+func generateEventID() int64 {
+	maxUint64 := ^uint64(0)
```
Comment (czeslavo) [inline tracing.go]: If I'm not wrong there are constants defined in `math` module: `MaxUint64`, `MinInt64`, you can consider using them here.

File: tracing.go (hunk 2 above)
Comment (krzysztofreczek) [reply tracing.go]: I will double-check that. In fact, the two top ones are context keys, while the third one is the span attribute key. The conventions may be different. Will get back with the results.

File: tracing.go (hunk 3 above)
Comment (krzysztofreczek) [reply tracing.go]: https://github.com/czeslavo/watermill-opencensus/pull/6/commits/0c281d9a162452456cbca755e765cb33add72009

File: tracing.go (hunk 2 above)
Comment (krzysztofreczek) [reply tracing.go]: https://github.com/czeslavo/watermill-opencensus/pull/6/commits/0c281d9a162452456cbca755e765cb33add72009

File: tracing.go (hunk 1 above)
Comment (krzysztofreczek) [reply tracing.go]: https://github.com/czeslavo/watermill-opencensus/pull/6/commits/0c281d9a162452456cbca755e765cb33add72009

**Review by czeslavo — APPROVED** [review]
🏅

**Pull request merged**